assignment/scripts/function-practice.js

```diff
@@ -1,79 +1,96 @@
-console.log('***** Function Practice *****')
+console.log("***** Function Practice *****");
 
 // Add the required code to complete the functions below
 // After _each_ function, use a console log to call the function
 // to test it and display the result
 
 // 1. Function to return 'Hello World!'
 function hello() {
-  return 'Hello World!';
+  return "Hello World!";
 }
 // Call the function to test
-console.log('Test - should say "Hello World!"', hello());
-
+console.log('Test - should say "Hello World!"', hello()); // moved this back to the original!
 
 // 2. Function to return an personalized hello, using the `name` argument.
 //    for example 'Hello, Jo!', or 'Hello, Stacy!'
-function helloName( name ) {
-  return;
-}
+function helloName(name) {
+  return console.log(`Hello, ${name}!`);
+} //
 // Remember to call the function to test
-
+helloName("David");
+helloName("Rebecca"); // Outputted both names correctly! Win! Stuck in term!
 
 // 3. Function to add two numbers together & return the result
-function addNumbers( firstNumber ) {
-  // return firstNumber + secondNumber;
+
+function addNumbers(num1, num2) {
+  let sum = num1 + num2;
+  return sum;
 }
 
+console.log(addNumbers(2, 5)); // followed this in class! Stuck again
 
 // 4. Function to multiply three numbers & return the result
-function multiplyThree( ){
-
+function multiplyThree(num1, num2, num3) {
+  let multiplication = num1 * num2 * num3;
+  return multiplication;
 }
 
+console.log(multiplyThree(2, 2, 3)); // output was 12
 
-// 5. Function that will return true if a number is positive, 
+// 5. Function that will return true if a number is positive,
 //    or greater than zero, and false otherwise
-function isPositive( number ) {
-  if ( number > 0 ){
-    return;
+function isPositive(number) {
+  if (number > 0) {
+    return true;
   }
-    return;
+  return false;
 }
-// Call the function to test each outcome (true & false) 
+// Call the function to test each outcome (true & false)
 // Write a separate console.log statement for each outcome
-console.log( 'isPositive - should say true', isPositive(3) );
-console.log( 'isPositive - should say false', isPositive(0) );
-console.log( 'isPositive - should say false', isPositive(-3) );
+console.log("isPositive - should say true", isPositive(3));
+console.log("isPositive - should say false", isPositive(0));
+console.log("isPositive - should say false", isPositive(-3));
 
+//We did this one pretty much in class!
 
-// 6. Function to return the _last_ item in an array. If the 
+// 6. Function to return the _last_ item in an array. If the
 //    array is empty, return `undefined`.
-function getLast( array ) {
-
+function getLast(array) {
+  if (array == []) {
```

Comment:
This if block will never trigger, because objects (including arrays) are compared by "identity" not value. In other words, they are only equal if they are literally the exact same object/array in memory. Since you are creating a new array here, it is guaranteed not to be equal to whatever array gets passed in. What you want to do here is instead check the length of the array (which is a number).

Thankfully, Line 62 works just fine on its own without this if block ever running.